File: src/euclidean_distance_client.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

using namespace std::chrono_literals;

int main(int argc, char const *argv[])
{   
    rclcpp::init(argc, argv);

    if(argc != 3)
    {
        RCLCPP_INFO(rclcpp::get_logger("Usage"),  "Input two coordinates");
        return 1;
    }
    
    std::shared_ptr<rclcpp::Node> node = rclcpp::Node::make_shared("euclidean_distance_client");
    rclcpp::Client<example_interfaces::srv::AddTwoInts>::SharedPtr client =   
        node->create_client<example_interfaces::srv::AddTwoInts>("dist_two_point");

    auto request = std::make_shared<example_interfaces::srv::AddTwoInts::Request>();

    request->a = atoll(argv[1]);
    request->b = atoll(argv[2]);

    while(!client->wait_for_service(1s))
    {
        if(!rclcpp::ok()){
            RCLCPP_ERROR(rclcpp::get_logger("Usage"), "Interupted while waiting service: Exiting");

            return 0;
        }
        RCLCPP_INFO(rclcpp::get_logger("Usage"),  "Service not availabe try again");
    }

    auto result = client->async_send_request(request);
    if(rclcpp::spin_until_future_complete(node, result) == rclcpp::FutureReturnCode::SUCCESS)
    {
        RCLCPP_INFO(rclcpp::get_logger("Usage"),  "Distance betweed Pose %ld", result.get()->sum);
        
    }else{
        RCLCPP_INFO(rclcpp::get_logger("Usage"),  "Failed to call service");
    }

    rclcpp::shutdown();
    return 0;
}

File: scripts/euclidean_distance_client.py
#!/usr/bin/env python3

import sys
import rclpy
from rclpy.node import Node
from example_interfaces.srv import AddTwoInts

class Client(Node):
    def __init__(self):
        super().__init__("euclidean_distance_client")

        self.client = self.create_client(AddTwoInts, "dist_two_point")

        while not self.client.wait_for_service(1):
            if not rclpy.ok():
                self.get_logger().info("Interupted while waiting service: Exiting")
                return 0
            else:
                self.get_logger().info("Service not availabe try again")

        self.request = AddTwoInts.Request()

    def send_request(self):
        self.request.a = int(sys.argv[1])
        self.request.b = int(sys.argv[2])
            
        self.result = self.client.call_async(request=self.request)



def main(args=None):
    rclpy.init(args=args)

    if(len(sys.argv) != 3):
        print("Input two coordinates")
        sys.exit()

    dist_client = Client()
    dist_client.send_request()

    while rclpy.ok():
        rclpy.spin_once(dist_client)
        if dist_client.result.done():
            try:
                response = dist_client.result.result()
            except Exception as e:
                dist_client.get_logger().info("Failed to call service {}".format(e))
            else:
                dist_client.get_logger().info("Distance between Pose {}".format(response.sum))
            break

    dist_client.destroy_node()
    rclpy.shutdown()

if __name__ == "__main__":
    main()    
    